src/Lib/Modes/PathEditMode.vala

```diff
@@ -50,10 +72,28 @@ public class Akira.Lib.Modes.PathEditMode : AbstractInteractionMode {
         live_points = new Geometry.Point[4];
         live_command = Type.LINE;
         live_idx = -1;
+        mode = Submode.APPEND;
     }
 
-    public void toggle_functionality (bool is_edit_path) {
-        this.is_edit_path = is_edit_path;
+    public void toggle_functionality (bool is_append_path) {
+        mode = is_append_path ? Submode.APPEND : Submode.EDIT;
+
+        // If we are editing the path, we need to set the first point in edit_model
+        if (mode == Submode.EDIT) {
+            double center_x = instance.components.center.x;
+            double center_y = instance.components.center.y;
+            double width = instance.components.size.width;
+            double height = instance.components.size.height;
+
+            var first_point = Geometry.Point (center_x - width / 2.0, center_y - height / 2.0);
+
+            double rotation = instance.components.transform.rotation;
```

Comment:
I mentioned this on element--but you can't do this. You have to apply the instance's transformation, not just the rotation. It is _never_ right to just apply rotation, You can test this by looking at transformed paths and you will see that editing is completely broken.